src/values/call_site_value.rs

```diff
@@ -220,12 +226,62 @@ impl<'ctx> CallSiteValue<'ctx> {
     ///
     /// let call_site_value = builder.build_call(fn_value, &[], "my_fn").unwrap();
     ///
-    /// assert_eq!(call_site_value.get_called_fn_value(), fn_value);
+    /// assert_eq!(call_site_value.get_called_fn_value(), Some(fn_value));
     /// ```
-    pub fn get_called_fn_value(self) -> FunctionValue<'ctx> {
-        use llvm_sys::core::LLVMGetCalledValue;
+    pub fn get_called_fn_value(self) -> Option<FunctionValue<'ctx>> {
+	// SAFETY: the passed LLVMValueRef is of type CallSite
+	let called_value = unsafe { LLVMGetCalledValue(self.as_value_ref()) };
+
+	let fn_value = unsafe { FunctionValue::new(called_value) };
+
+	// Check that the retrieved function value has the same type as the callee.
+	// This matches the behavior of the C++ API `CallBase::getCalledFunction`.
+	// This is only possible on LLVM >=8, where the `LLVMGetCalledFunctionType` API exists.
+	self.get_called_fn_value_check_type_consistency(fn_value)
+    }
+
+    #[llvm_versions(..8)]
+    #[inline]
+    fn get_called_fn_value_check_type_consistency(&self, fn_value: Option<FunctionValue<'ctx>>) -> Option<FunctionValue<'ctx>> {
+	fn_value
+    }
+
+    #[llvm_versions(8..)]
+    #[inline]
+    fn get_called_fn_value_check_type_consistency(&self, fn_value: Option<FunctionValue<'ctx>>) -> Option<FunctionValue<'ctx>> {
+	fn_value.filter(|fn_value| fn_value.get_type() == self.get_called_fn_type())
```

Comment:
Please use cargo fmt on your changes... it appears to be mixing in tabs and looks off

Reply:
Thanks for the feedback! I fixed formatting issues related to the changes in this PR. Other changes applied by `rustfmt` are in #575.